lib/matplotlib/axes/_axes.py

```diff
@@ -1997,20 +1998,14 @@ def bar(self, left, height, width=0.8, bottom=None, **kwargs):
         label = kwargs.pop('label', '')
         tick_labels = kwargs.pop('tick_label', None)
 
-        def make_iterable(x):
-            if not iterable(x):
-                return [x]
-            else:
-                return x
-
         # make them safe to take len() of
         _left = left
-        left = make_iterable(left)
```

Comment:
Also, note that the current implementation is actually buggy: `bar([1, 2], [3, 4], width=[.8])` works but `bar([1, 2], [3, 4], width=np.array([.8]))` currently raises an exception due to the fact that `width *= nbars` works elementwise here.  (Can you open a separate issue if you don't think this PR can get merged any time soon, so that it doesn't get lost?)